experiment/src/org/labkey/experiment/ExpDataIterators.java

```
@@ -1807,9 +1810,22 @@ public boolean next() throws BatchValidationException
 
             if (hasNext)
             {
-                String lsid = (String) get(_lsidCol);
-                if (null != lsid)
-                    _lsids.add(lsid);
+                if (_useExistingRecord)
+                {
+                    Map<String, Object> map = getExistingRecord();
+                    if (map != null && map.containsKey("lsid"))
+                    {
+                        String lsid = (String) map.get("lsid");
+                        if (null != lsid)
+                            _lsids.add(lsid);
```

Should we throw an error if we don't have an lsid to add at this point or if we don't get a map for he existing record? I'm not sure when that would happen, but it seems like a problem of some sort.

---

I think there is argument for either way...If the update itself succeed but indexing is skipped it's probably fine?

---

Yes, I think that's true.